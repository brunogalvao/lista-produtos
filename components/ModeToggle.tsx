"use client";

import { useTheme } from "next-themes";
import { Moon, Sun } from "lucide-react";
import { Button } from "@/components/ui/button";
import { useEffect, useState } from "react";

export function ModeToggle() {
  const { setTheme, theme } = useTheme();
  const [mounted, setMounted] = useState(false);

  useEffect(() => {
    setMounted(true);
  }, []);

  if (!mounted) return null;

  return (
    <Button
      variant="outline"
      onClick={() => setTheme(theme === "dark" ? "light" : "dark")}
      aria-label="Alternar tema"
      className={
        theme === "dark"
          ? "bg-zinc-500 text-amber-400"
          : "bg-zinc-800 text-zinc-100"
      }
    >
      {theme === "dark" ? (
        <>
          <span>Light</span>
          <Sun className="h-5 w-5" />
        </>
      ) : (
        <>
          <span>Dark</span>
          <Moon className="h-5 w-5" />
        </>
      )}
    </Button>
  );
}
