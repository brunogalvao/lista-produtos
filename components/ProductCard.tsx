"use client";

import { Product } from "@/types/product";
import { formatToBRL } from "@/utils/format";
import Image from "next/image";
import {
  Card,
  CardContent,
  CardFooter,
  CardHeader,
  CardTitle,
} from "./ui/card";
import { Label } from "./ui/label";
import { Badge } from "./ui/badge";

type Props = {
  product: Product;
};

export function ProductCard({ product }: Props) {
  return (
    <Card>
      <CardHeader>
        <CardTitle>{product.name}</CardTitle>
      </CardHeader>
      <CardContent>
        <div className="flex flex-col gap-3">
          <div className="relative w-full aspect-[3/2]">
            <Image
              src={product.image}
              alt={product.name}
              fill
              className="object-contain object-center rounded border"
              sizes="(max-width: 768px) 100vw, 33vw"
            />
          </div>

          <div className="flex flex-col">
            <Label className="text-[10px] text-foreground">Categoria</Label>
            <span className="text-[12px] text-foreground">
              <Badge variant="default" className="dark:text-foreground">
                {product.category}
              </Badge>
            </span>
          </div>

          <div className="flex flex-col space-y-2">
            <Label className="text-[10px] text-foreground">Descrição</Label>
            <div className="text-base text-foreground line-clamp-2 dark:bg-zinc-800 bg-zinc-100 p-2 h-20 rounded">
              {product.description}
            </div>
          </div>
        </div>
      </CardContent>

      <CardFooter>
        <div className="mt-auto w-full font-bold bg-green-600 text-white p-3 rounded-full text-center">
          {formatToBRL(product.price)}
        </div>
      </CardFooter>
    </Card>
  );
}
