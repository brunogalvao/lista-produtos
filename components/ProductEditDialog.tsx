"use client";

import { useState } from "react";
import {
  Dialog,
  DialogTrigger,
  DialogContent,
  DialogHeader,
  DialogTitle,
  DialogFooter,
} from "@/components/ui/dialog";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Product } from "@/types/product";
import { Pencil } from "lucide-react";

type Props = {
  product: Product;
  onSave: (updatedProduct: Product) => void;
};

export function ProductEditDialog({ product, onSave }: Props) {
  const [open, setOpen] = useState(false);
  const [form, setForm] = useState<Product>(product);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setForm({ ...form, [e.target.name]: e.target.value });
  };

  const handleSave = () => {
    onSave({ ...form, price: Number(form.price) });
    setOpen(false);
  };

  return (
    <Dialog open={open} onOpenChange={setOpen}>
      <DialogTrigger asChild>
        <Button variant="outline" size="sm">
          <Pencil />
        </Button>
      </DialogTrigger>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Editar Produto</DialogTitle>
        </DialogHeader>

        <div className="space-y-3">
          <Input
            name="name"
            placeholder="Nome"
            value={form.name}
            onChange={handleChange}
          />
          <Input
            name="category"
            placeholder="Categoria"
            value={form.category}
            onChange={handleChange}
          />
          <Input
            name="price"
            placeholder="PreÃ§o"
            type="number"
            value={form.price}
            onChange={handleChange}
          />
        </div>

        <DialogFooter className="mt-4">
          <Button onClick={handleSave} className=" dark:text-foreground">
            Salvar
          </Button>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  );
}
