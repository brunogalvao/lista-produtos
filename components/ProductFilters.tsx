"use client";

import { useState } from "react";
import { useProductStore } from "@/store/productStore";
import { Product } from "@/types/product";
import { getAllProducts } from "@/lib/getAllProducts";
import { Input } from "./ui/input";
import { Button } from "./ui/button";
import { Label } from "./ui/label";
import { Card, CardContent, CardHeader, CardTitle } from "./ui/card";
import { BrushCleaning, Funnel } from "lucide-react";

export default function ProductFilter() {
  const { setProducts } = useProductStore();
  const [search, setSearch] = useState("");
  const [minPrice, setMinPrice] = useState("");
  const [maxPrice, setMaxPrice] = useState("");
  const [sortBy, setSortBy] = useState(""); // novo

  const handleFilter = async () => {
    const res = await fetch("/api/products");
    const allProducts: Product[] = await res.json();

    const filtered = allProducts.filter((product) => {
      const matchesName = product.name
        .toLowerCase()
        .includes(search.toLowerCase());

      const price = product.price;
      const min = minPrice ? parseFloat(minPrice) : 0;
      const max = maxPrice ? parseFloat(maxPrice) : Infinity;
      const matchesPrice = price >= min && price <= max;

      return matchesName && matchesPrice;
    });

    const sorted = [...filtered].sort((a, b) => {
      switch (sortBy) {
        case "name-asc":
          return a.name.localeCompare(b.name);
        case "name-desc":
          return b.name.localeCompare(a.name);
        case "price-asc":
          return a.price - b.price;
        case "price-desc":
          return b.price - a.price;
        default:
          return 0;
      }
    });

    setProducts(filtered);
    setProducts(sorted);
  };

  const handleReset = () => {
    const all = getAllProducts();
    setProducts(all);
    setSearch("");
    setMinPrice("");
    setMaxPrice("");
    setSortBy("");
  };

  return (
    <Card>
      <CardHeader>
        <CardTitle>Filtro</CardTitle>
      </CardHeader>

      <CardContent>
        <div className="flex flex-col sm:flex-row items-end gap-3 mb-6">
          <div className="flex flex-col w-full space-y-1">
            <Label className="text-sm">Nome</Label>
            <Input
              type="text"
              placeholder="Buscar por nome"
              value={search}
              onChange={(e) => setSearch(e.target.value)}
            />
          </div>
          <div className="flex flex-col w-full space-y-1">
            <Label className="text-sm">Preço Minimo</Label>
            <Input
              type="number"
              placeholder="Preço mínimo"
              value={minPrice}
              onChange={(e) => setMinPrice(e.target.value)}
            />
          </div>
          <div className="flex flex-col w-full space-y-1">
            <Label className="text-sm">Preço Maximo</Label>
            <Input
              type="number"
              placeholder="Preço máximo"
              value={maxPrice}
              onChange={(e) => setMaxPrice(e.target.value)}
            />
          </div>
        </div>

        <div className="flex justify-end">
          <div className="flex flex-row gap-3 md:w-[50%] w-full">
            <div className="w-full">
              <Button
                onClick={handleFilter}
                variant="default"
                className="dark:text-foreground w-full"
              >
                Filtrar
                <Funnel />
              </Button>
            </div>

            <div className="w-full">
              <Button
                onClick={handleReset}
                variant="destructive"
                className="w-full"
              >
                Resetar
                <BrushCleaning />
              </Button>
            </div>
          </div>
        </div>
      </CardContent>
    </Card>
  );
}
