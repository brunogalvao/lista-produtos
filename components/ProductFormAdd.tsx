"use client";

import { useState } from "react";
import { useProductStore } from "@/store/productStore";
import { Product } from "@/types/product";

import {
  Dialog,
  DialogTrigger,
  DialogContent,
  DialogHeader,
  DialogTitle,
  DialogDescription,
} from "@/components/ui/dialog";

import { Plus } from "lucide-react";
import { Button } from "./ui/button";
import { Input } from "./ui/input";
import { Label } from "./ui/label";

export default function ProductFormAdd() {
  const { setProducts } = useProductStore();

  const [open, setOpen] = useState(false);

  const [form, setForm] = useState({
    name: "",
    price: "",
    description: "",
    category: "",
    image: "",
  });

  const handleChange = (
    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
  ) => {
    setForm({ ...form, [e.target.name]: e.target.value });
  };

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();

    const newProduct = {
      id: Date.now().toString(),
      name: form.name,
      price: parseFloat(form.price),
      description: form.description,
      category: form.category || "Outro",
      image: form.image,
    };

    try {
      const res = await fetch("/api/products", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(newProduct),
      });

      const data = await res.json();
      setProducts((prev: Product[]) => [data, ...prev]);

      // limpa o formulário
      setForm({
        name: "",
        price: "",
        description: "",
        category: "",
        image: "",
      });

      setOpen(false);
    } catch (err) {
      console.error("Erro ao adicionar produto:", err);
    }
  };

  return (
    <Dialog open={open} onOpenChange={setOpen}>
      <DialogTrigger asChild>
        <Button variant="default" className="dark:text-foreground">
          Adicionar Produto
          <Plus className="inline mr-2" size={16} />
        </Button>
      </DialogTrigger>

      <DialogContent className="max-w-md">
        <DialogHeader>
          <DialogTitle>Adicionar Produto</DialogTitle>
          <DialogDescription>
            Preencha os campos abaixo para adicionar um novo produto. Para a
            imagem, você pode copiar o link de uma imagem diretamente do site{" "}
            <a
              href="https://unsplash.com"
              target="_blank"
              rel="noopener noreferrer"
              className="hover:underline text-foreground"
            >
              unsplash.com
            </a>{" "}
            e colá-lo no campo{" "}
            <span
              dangerouslySetInnerHTML={{ __html: "&ldquo;Imagem (URL)&rdquo;" }}
            />
            .
          </DialogDescription>
        </DialogHeader>

        <form onSubmit={handleSubmit} className="flex flex-col gap-4 mt-4">
          <div className="flex space-y-2 flex-col">
            <Label className="text-sm">Nome</Label>
            <Input
              type="text"
              name="name"
              value={form.name}
              onChange={handleChange}
              placeholder="Nome do produto"
              className="border rounded px-3 py-2 w-full"
              required
            />
          </div>

          <div className="flex space-y-2 flex-col">
            <Label className="text-sm">Categoria</Label>
            <Input
              type="text"
              name="category"
              value={form.category}
              onChange={handleChange}
              placeholder="Categoria"
              className="border rounded px-3 py-2 w-full"
              required
            />
          </div>

          <div className="flex space-y-2 flex-col">
            <Label className="text-sm">Preço</Label>
            <Input
              type="number"
              name="price"
              value={form.price}
              onChange={handleChange}
              placeholder="Preço"
              className="border rounded px-3 py-2 w-full"
              required
            />
          </div>

          <div className="flex space-y-2 flex-col">
            <Label className="text-sm">Descrição</Label>
            <textarea
              name="description"
              value={form.description}
              onChange={handleChange}
              placeholder="Descrição"
              className="border rounded px-3 py-2 w-full"
              required
            />
          </div>

          <div className="flex space-y-2 flex-col">
            <Label className="text-sm">Imagem (URL)</Label>
            <input
              type="url"
              name="image"
              value={form.image}
              onChange={handleChange}
              placeholder="https://..."
              className="border rounded px-3 py-2 w-full"
              required
            />
          </div>

          <Button type="submit" variant="default" className="text-foreground">
            Cadastrar Produto
          </Button>
        </form>
      </DialogContent>
    </Dialog>
  );
}
