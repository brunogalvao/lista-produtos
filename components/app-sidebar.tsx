"use client";

import { Home, Inbox } from "lucide-react";
import Link from "next/link";
import { usePathname } from "next/navigation";
import { cn } from "@/lib/utils";
import { ModeToggle } from "./ModeToggle";

import {
  Sidebar,
  SidebarContent,
  SidebarFooter,
  SidebarGroup,
  SidebarGroupContent,
  SidebarGroupLabel,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
} from "@/components/ui/sidebar";

const items = [
  {
    title: "Lista de Produtos",
    url: "/",
    icon: Home,
  },
  {
    title: "Gerenciar Produtos",
    url: "/gerenciar",
    icon: Inbox,
  },
];

export function AppSidebar() {
  const pathname = usePathname();

  return (
    <Sidebar>
      <SidebarContent>
        <SidebarGroup>
          <SidebarGroupLabel>Application</SidebarGroupLabel>
          <SidebarGroupContent>
            <SidebarMenu>
              {items.map((item) => {
                const isActive = pathname === item.url;

                return (
                  <SidebarMenuItem key={item.title}>
                    <SidebarMenuButton asChild>
                      <Link
                        href={item.url}
                        className={cn(
                          "flex items-center gap-2 h-10 rounded-md hover:bg-muted transition-all",
                          isActive && "bg-muted font-semibold",
                        )}
                      >
                        <item.icon className="w-4 h-4" />
                        <span>{item.title}</span>
                      </Link>
                    </SidebarMenuButton>
                  </SidebarMenuItem>
                );
              })}
            </SidebarMenu>
          </SidebarGroupContent>
        </SidebarGroup>
      </SidebarContent>

      <SidebarFooter>
        <div className="pb-10 flex justify-center">
          <ModeToggle />
        </div>
      </SidebarFooter>
    </Sidebar>
  );
}
