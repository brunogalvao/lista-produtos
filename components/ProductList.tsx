"use client";

import { useEffect, useState } from "react";
import { useProductStore } from "@/store/productStore";
import { ProductCard } from "@/components/ProductCard";
import { useProductCount } from "@/context/ProductCountContext";
import { ChevronLeft, ChevronRight } from "lucide-react";
import { Button } from "./ui/button";

const ITEMS_PER_PAGE = 6;

export default function ProductList() {
  const { products, setProducts } = useProductStore();
  const { setCount } = useProductCount();

  const [currentPage, setCurrentPage] = useState(1);

  useEffect(() => {
    if (typeof window === "undefined") return;

    const run = async () => {
      const res = await fetch("/api/products");
      try {
        const data = await res.json();
        setProducts(data);
      } catch (err) {
        console.error("Erro ao carregar produtos:", err);
      }
    };

    run();
  }, [setProducts]);

  useEffect(() => {
    setCount(products.length);
  }, [products, setCount]);

  const totalPages = Math.ceil(products.length / ITEMS_PER_PAGE);
  const start = (currentPage - 1) * ITEMS_PER_PAGE;

  const sortedProducts = [...products].sort((a, b) =>
    a.name.localeCompare(b.name),
  );

  const paginatedProducts = sortedProducts.slice(start, start + ITEMS_PER_PAGE);

  const goToPrevious = () => {
    if (currentPage > 1) setCurrentPage((prev) => prev - 1);
  };

  const goToNext = () => {
    if (currentPage < totalPages) setCurrentPage((prev) => prev + 1);
  };

  return (
    <div className="space-y-6">
      <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-4">
        {paginatedProducts.map((product) => (
          <ProductCard key={product.id} product={product} />
        ))}
      </div>

      {/* Paginação */}
      <div className="flex justify-center items-center gap-4">
        <Button
          variant="outline"
          onClick={goToPrevious}
          disabled={currentPage === 1}
        >
          <ChevronLeft />
        </Button>

        <span className="text-foreground text-sm">
          Página {currentPage} de {totalPages}
        </span>

        <Button
          variant="outline"
          onClick={goToNext}
          disabled={currentPage === totalPages}
        >
          <ChevronRight />
        </Button>
      </div>
    </div>
  );
}
