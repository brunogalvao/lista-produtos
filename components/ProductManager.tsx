"use client";

import { useEffect, useState } from "react";
import {
  Table,
  TableBody,
  TableCaption,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "@/components/ui/table";
import { Button } from "@/components/ui/button";
import { Pencil, Trash } from "lucide-react";
import { Product } from "@/types/product";
import { formatToBRL } from "@/utils/format";
import { ProductEditDialog } from "./ProductEditDialog";
import { getStoredProducts, saveStoredProducts } from "@/lib/storage";
import { Card, CardContent, CardHeader, CardTitle } from "./ui/card";
import { toast } from "sonner";

export default function ProductManager() {
  const [products, setProducts] = useState<Product[]>([]);

  useEffect(() => {
    const products = getStoredProducts();
    setProducts(products);
  }, []);

  const handleDelete = (id: string) => {
    const updated = products.filter((p) => p.id !== id);
    setProducts(updated);
    saveStoredProducts(updated);

    toast.warning("Produto excluído com sucesso!");
  };

  const handleSaveEdit = (updated: Product) => {
    const updatedList = products.map((p) =>
      p.id === updated.id ? updated : p,
    );

    setProducts(updatedList);
    saveStoredProducts(updatedList);

    toast.success("Produto atualizado com sucesso!");
  };

  return (
    <Card>
      <CardHeader>
        <CardTitle>Gerenciador de Produtos</CardTitle>
      </CardHeader>
      <CardContent>
        <div className="p-6 space-y-4">
          {products.length === 0 ? (
            <p className="text-muted-foreground">Nenhum produto encontrado.</p>
          ) : (
            <Table>
              <TableCaption>
                Lista de todos os produtos cadastrados
              </TableCaption>
              <TableHeader>
                <TableRow>
                  <TableHead>Nome</TableHead>
                  <TableHead>Categoria</TableHead>
                  <TableHead>Preço</TableHead>
                  <TableHead className="w-[150px] text-center">Ações</TableHead>
                </TableRow>
              </TableHeader>
              <TableBody>
                {products.map((product) => (
                  <TableRow key={product.id}>
                    <TableCell className="font-medium">
                      {product.name}
                    </TableCell>
                    <TableCell>{product.category}</TableCell>
                    <TableCell>{formatToBRL(product.price)}</TableCell>
                    <TableCell className="flex justify-center gap-2">
                      <ProductEditDialog
                        product={product}
                        onSave={handleSaveEdit}
                      />
                      <Button
                        variant="destructive"
                        size="sm"
                        onClick={() => handleDelete(product.id)}
                      >
                        <Trash className="w-4 h-4" />
                      </Button>
                    </TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          )}
        </div>
      </CardContent>
    </Card>
  );
}
