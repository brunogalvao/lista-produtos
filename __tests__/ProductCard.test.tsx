import { render } from "@testing-library/react";
import { ProductCard } from "@/components/ProductCard";
import { Product } from "@/types/product";

const mockProduct: Product = {
  id: "1",
  name: "Produto Teste",
  category: "Categoria Teste",
  price: 99.99,
  description: "Descrição teste",
  image: "https://images.unsplash.com/photo-xyz",
};

describe("ProductCard snapshot", () => {
  it("deve bater com o snapshot", () => {
    const { container } = render(<ProductCard product={mockProduct} />);
    expect(container).toMatchSnapshot();
  });
});
