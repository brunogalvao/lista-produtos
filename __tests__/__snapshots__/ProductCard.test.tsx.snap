// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductCard snapshot deve bater com o snapshot 1`] = `
<div>
  <div
    class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm"
    data-slot="card"
  >
    <div
      class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
      data-slot="card-header"
    >
      <div
        class="leading-none font-semibold"
        data-slot="card-title"
      >
        Produto Teste
      </div>
    </div>
    <div
      class="px-6"
      data-slot="card-content"
    >
      <div
        class="flex flex-col gap-3"
      >
        <div
          class="relative w-full aspect-[3/2]"
        >
          <img
            alt="Produto Teste"
            class="object-contain object-center rounded border"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="(max-width: 768px) 100vw, 33vw"
            src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-xyz&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col"
        >
          <label
            class="flex items-center gap-2 font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-[10px] text-foreground"
            data-slot="label"
          >
            Categoria
          </label>
          <span
            class="text-[12px] text-foreground"
          >
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-primary text-primary-foreground [a&]:hover:bg-primary/90 dark:text-foreground"
              data-slot="badge"
            >
              Categoria Teste
            </span>
          </span>
        </div>
        <div
          class="flex flex-col space-y-2"
        >
          <label
            class="flex items-center gap-2 font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-[10px] text-foreground"
            data-slot="label"
          >
            Descrição
          </label>
          <div
            class="text-base text-foreground line-clamp-2 dark:bg-zinc-800 bg-zinc-100 p-2 h-20 rounded"
          >
            Descrição teste
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center px-6 [.border-t]:pt-6"
      data-slot="card-footer"
    >
      <div
        class="mt-auto w-full font-bold bg-green-600 text-white p-3 rounded-full text-center"
      >
        R$ 99,99
      </div>
    </div>
  </div>
</div>
`;
